### PR TITLE
DNSLink update - targeted grant

### PR DESCRIPTION
**Pull request by martinheidegger**
Hello,

Here's the targeted devgrant for updating dnslink and it's implementations

CC @autonome @lidel

### CODE REVIEW

File: targeted-grants/dnslink-multiplatform-update.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+## Acceptance Criteria
+
+<!-- What are the acceptance criteria for each milestone and for the final deliverables? These should be as objective as possible. They will be used to determine whether or not a grantee will receive payment for work completed for a milestone. -->
+
+1. Published Test Setup and examplary use of it in one implementation.
+2. Pull-Request on dnslink homepage containing all required changes to the spec.
+3. JavaScript library with reference implementation of the DNSLink spec published on NPM. Programmatic interface for resolving DNSLink records works in Nodejs and Browser, and command line script can be used for quick lookup/validation when run via `npx`.
+4. Browser Util, based on the NPM library, published on the dnslink.io website allowing to explore and test DNSLink records on a domain provided by the user. DoH endpoint used for DNS lookups can be customized via UI.
+6. Accepted PR's to existing IPFS repositories. go-ipfs uses go-dnslink, and js-ipfs uses the new dnslink library from NPM.
```
Comment (lidel) [inline targeted-grants/dnslink-multiplatform-update.md:74]:
```suggestion
6. Accepted PR's to existing IPFS repositories. go-ipfs uses go-dnslink, and js-ipfs uses the new dnslink library from NPM. 
7. New libraries will be released under [dual MIT/Apache-2 license](https://protocol.ai/blog/announcing-the-permissive-license-stack/)
```

**Review by lidel — APPROVED** [review]
LGTM.  

@martinheidegger btw: as this may produce some new projects, let's make sure we use dual license for them, to maximize the utility o DNSLink standard (added below).

@autonome mind reviewing + what is the next step here?

File: targeted-grants/dnslink-multiplatform-update.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+| 4 | Browser util | $1300 | ~20h |
+| 5 | Fixes for the existing implementations | $2600 | ~40h  |
+
+## Acceptance Criteria
+
+<!-- What are the acceptance criteria for each milestone and for the final deliverables? These should be as objective as possible. They will be used to determine whether or not a grantee will receive payment for work completed for a milestone. -->
+
+1. Published Test Setup and examplary use of it in one implementation.
+2. Pull-Request on dnslink homepage containing all required changes to the spec.
+3. JavaScript library with reference implementation of the DNSLink spec published on NPM. Programmatic interface for resolving DNSLink records works in Nodejs and Browser, and command line script can be used for quick lookup/validation when run via `npx`.
```
Comment (lidel) [inline targeted-grants/dnslink-multiplatform-update.md:72]: (CLI script is here)

**Comment by autonome** [comment]
Thanks for working on this grant! Please create a discussion thread in the format of `[GRANT] Your Grant Title` in the community repo and provide updates on your milestones on a weekly basis: https://github.com/ipfs/community/discussions

**Comment by martinheidegger** [comment]
Thank you @autonome - Discussion thread opened: https://github.com/ipfs/community/discussions/560

**Pull request merged**